addons/pos_mercado_pago/models/pos_payment_method.py

```diff
@@ -0,0 +1,113 @@
+        """
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Do not have access to fetch token from Mercado Pago"))
+
+        mercado_pago = MercadoPagoPosRequest(self.mp_bearer_token)
+        # Call Mercado Pago for payment intend creation
+        resp = mercado_pago._call_mercado_pago("post", f"/point/integration-api/devices/{self.mp_id_point_smart}/payment-intents", infos)
+        _logger.debug("mp_payment_intent_create(), response from Mercado Pago: %s", resp)
+        return resp
+
+    def mp_payment_intent_get(self, payment_intent_id):
+        """
+        Called from frontend to get the last payment intend from Mercado Pago
+        """
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Do not have access to fetch token from Mercado Pago"))
+
+        mercado_pago = MercadoPagoPosRequest(self.mp_bearer_token)
+        # Call Mercado Pago for payment intend status
+        resp = mercado_pago._call_mercado_pago("get", f"/point/integration-api/payment-intents/{payment_intent_id}", {})
+        _logger.debug("mp_payment_intent_get(), response from Mercado Pago: %s", resp)
+        return resp
+
+    def mp_payment_intent_cancel(self, payment_intent_id):
+        """
+        Called from frontend to cancel a payment intent in Mercado Pago
+        """
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Do not have access to fetch token from Mercado Pago"))
+
+        mercado_pago = MercadoPagoPosRequest(self.mp_bearer_token)
+        # Call Mercado Pago for payment intend cancelation
+        resp = mercado_pago._call_mercado_pago("delete", f"/point/integration-api/devices/{self.mp_id_point_smart}/payment-intents/{payment_intent_id}", {})
+        _logger.debug("mp_payment_intent_cancel(), response from Mercado Pago: %s", resp)
+        return resp
+
+    def _find_terminal(self, token, point_smart):
+        mercado_pago = MercadoPagoPosRequest(token)
+        data = mercado_pago._call_mercado_pago("get", "/point/integration-api/devices", {})
+        if 'devices' in data:
+            # Search for a device id that contains the serial number entered by the user
+            found_device = next((device for device in data['devices'] if point_smart in device.get('id')), None)
+
+            if not found_device:
+                raise UserError(_("The terminal serial number is not registered on Mercado Pago"))
+
+            return found_device.get('id', '')
+        else:
+            raise UserError(_("Please verify your production user token as it was rejected"))
+
+    def write(self, vals):
+        if self.mp_bearer_token:
+            point_smart = vals.get('mp_id_point_smart', self.mp_id_point_smart)
+            if not 'name' in vals:
+                vals['mp_id_point_smart'] = self._find_terminal(vals.get('mp_bearer_token', self.mp_bearer_token), point_smart)
```

Comment:
First question: Yes, but I figured out that this logic is better:
`if any(elem in vals for elem in['mp_id_point_smart', 'mp_bearer_token']):`

Second question: each time user changes one (or the two) of the the two fields above, we have to recheck with Mercado data:
- user change the bearer_token: check if correct
- user change the id_point_smart: check to get the correct one

Reply:
To make things clearer but also to avoid infinite loop in the write I've added a new field "mp_id_point_smart_complet" that contains the complete id retrieved from Mercado Pago. It is used in the code instead of "mp_id_point_smart". The field "mp_id_point_smart" (the one entered by the user) is used only for seeking the complete one:
```
   def write(self, vals):
        records = super().write(vals)

        if 'mp_id_point_smart' in vals or 'mp_bearer_token' in vals:
            self.mp_id_point_smart_complet = self._find_terminal(self.mp_bearer_token, self.mp_id_point_smart)

        return records

```